test/Microsoft.AspNetCore.Server.Kestrel.TestCommon/PortManager.cs

```diff
@@ -0,0 +1,20 @@
+using Microsoft.AspNetCore.Server.Testing.Common;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.TestCommon
+{
+    public static class PortManager
+    {
+        public static int _nextPort = 8001;
+        public static object _portLock = new object();
+        
+        public static int GetNextPort()
```

Comment:
Just realized there's a problem with this logic. `TestUriHelper.FindFreePort` won't search from `_nextPort` and above if the initial port is not available. This defeats the purpose of this class.

Will submit an update soon.